ts/src/pro/blofin.ts

```diff
@@ -0,0 +1,489 @@
+        let stored = this.safeValue (this.ohlcvs[symbol], unifiedTimeframe);
+        if (stored === undefined) {
+            const limit = this.safeInteger (this.options, 'OHLCVLimit', 1000);
+            stored = new ArrayCacheByTimestamp (limit);
+            this.ohlcvs[symbol][unifiedTimeframe] = stored;
+        }
+        for (let i = 0; i < data.length; i++) {
+            const candle = data[i];
+            const parsed = this.parseOHLCV (candle, market);
+            stored.append (parsed);
+        }
+        const resolveData = [ symbol, unifiedTimeframe, stored ];
+        const messageHash = 'candle' + interval + ':' + symbol;
+        client.resolve (resolveData, messageHash);
+    }
+
+    handleMessage (client: Client, message) {
+        //
+        // message examples
+        //
+        // {
+        //   arg: {
+        //     channel: "trades",
+        //     instId: "DOGE-USDT",
+        //   },
+        //   event: "subscribe"
+        // }
+        //
+        // incoming data updates' examples can be seen under each handler method
+        //
+        const methods = {
+            'trades': this.handleWsTrades,
+            'books': this.handleWsOrderBook,
+            'tickers': this.handleWsTicker,
+            'candle': this.handleWsOHLCV,
+            'pong': this.handleWsPong,
+        };
+        let method = undefined;
+        if (message === 'pong') {
+            method = this.safeValue (methods, 'pong');
+        } else {
+            const event = this.safeString (message, 'event');
+            if (event === 'subscribe') {
+                return;
+            }
+            const arg = this.safeDict (message, 'arg');
+            const channelName = this.safeString (arg, 'channel');
+            method = this.safeValue (methods, channelName);
+            if (!method && channelName.indexOf ('candle') >= 0) {
+                method = methods['candle'];
+            }
+        }
+        if (method) {
+            method.call (this, client, message);
+        }
+    }
+
+    async watchMultipleWrapper (channelName: string, callerMethodName: string, symbolsArray, params = {}) {
+        // underlier method for all watch-multiple symbols
+        await this.loadMarkets ();
+        [ callerMethodName, params ] = this.handleParamString (params, 'callerMethodName', callerMethodName);
+        // if OHLCV method are being called, then symbols would be symbolsAndTimeframes (multi-dimensional) array
+        const isOHLCV = (channelName === 'candle');
+        let symbols = !isOHLCV ? symbolsArray : this.getListFromObjectValues (symbolsArray, 0);
+        let firstMarket = undefined;
+        symbols = this.marketSymbols (symbols, undefined, false, true);
+        if (symbols !== undefined) {
+            firstMarket = this.market (symbols[0]);
+        }
+        let marketType = undefined;
+        [ marketType, params ] = this.handleMarketTypeAndParams (callerMethodName, firstMarket, params);
```

Comment:
I think we can do `market['spot']` here, first blofin does not have conflicting ids and second it is swap only for now, so I don't know if we need this check here

Reply:
can you plz suggest a change? i think i exactly don't know which line you want to be amended into how